### PR TITLE
Update ray operator Dockerfile

### PR DESCRIPTION
**Pull request by anishasthana**
## Why are these changes needed?

We are currently using generic golang and distroless images to build the ray-operator (and other images).
Using “ubi(universal base image)” and “scratch” will improve the security of our images. [UBI](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image) images are actively maintained, scanned, and updated by Red Hat productization and security teams. Using these images will improve user confidence in what is being provided by the KubeRay community.

`scratch` is similar to the distroless, but this will make the dockerfile more portable for build systems that are unable to access `gcr.io/distroless/static`

## Related issue number

Not directly related, but we should really update the Go version we are using: https://github.com/ray-project/kuberay/issues/518

image

Go 1.17 seems to have a lot of vulnerabilities. I'll see if I have some time in the next couple weeks to update it.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

I'm happy to create an issue if required. cc @kevin85421

### CODE REVIEW

**Comment by anishasthana** [comment]
@kevin85421 can you retrigger CI here?

File: ray-operator/Dockerfile [hunk 1]
```diff
@@ -15,11 +15,10 @@ COPY apis/ apis/
 COPY controllers/ controllers/
 
 # Build
+USER root
```
Comment (kevin85421) [inline ray-operator/Dockerfile:18]: What prompted this change? Could it potentially raise any security concerns?

**Review by kevin85421 — COMMENTED** [review]
> make the dockerfile more portable for build systems that are unable to access gcr.io/distroless/static

Would you mind providing more details about why some systems are unable to access `gcr.io/distroless/static`? Thx

File: ray-operator/Dockerfile (hunk 1 above)
Comment (anishasthana) [reply ray-operator/Dockerfile:18]: This change was prompted by the switch over to `registry.access.redhat.com/ubi8/go-toolset:1.17.12-11`. The default user in the `go-toolset` image does not have the right permissions to be able to build the binary. It doesn't raise any security concerns as we are specifically setting a non-root user at https://github.com/ray-project/kuberay/pull/1213/files#diff-43d3bfc86d147402a0bff47e59e7aadb5e8a7e3154b32accce70f6e20a824b44R24.

**Comment by anishasthana** [comment]
>Would you mind providing more details about why some systems are unable to access gcr.io/distroless/static?
Some build systems are unable to pull images from public "non-trusted" registries, so gcr.io (or even quay/dockerhub) would be out of the question unless you go through approval processes. As `SCRATCH` is effectively equivalent to distroless and it doesn't require going to external registries, it would be a good substitute for users who need to build the kuberay operator image in their own internal environments.

**Comment by anishasthana** [comment]
@kevin85421 bump on this

**Review by kevin85421 — APPROVED** [review]
LGTM

**Pull request merged**